### PR TITLE
docs: revise default values and descriptions

### PR DESCRIPTION
**Pull request by Sarin-Udompanish**
## Description
Review content in API section in all elements

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: documents/scripts/element.injector.mjs [hunk 1]
```diff
@@ -41,23 +42,26 @@ const handler = async () => {
...
+            let nextIndex = i+1;
+            let nextTitleIndex = apiReferenceIndices[nextIndex];
+            while(nextTitleIndex === -1 && nextIndex < apiReferenceIndices.length) {
+              nextIndex++;
+              nextTitleIndex = apiReferenceIndices[nextIndex];
+            }
+            const lastIndex = nextTitleIndex && nextTitleIndex !== -1 ? nextTitleIndex - titleIndex : apiContent.length;
+            content += '\n#' + apiContent.substr(titleIndex, lastIndex) + '\n';
+            const isEventSection = i === 2;
+            if(isEventSection) {
```
Comment (wattachai-lseg) [inline documents/scripts/element.injector.mjs]: magic string provides more context than a magic number, right?
```suggestion
            if(API_REFERENCE_TITLES[i] === '## Events') {
```

File: documents/scripts/element.injector.mjs [hunk 2]
```diff
@@ -8,6 +8,7 @@ const PACKAGE_ROOT = '../node_modules/@refinitiv-ui/elements/lib';
 const ELEMENT_API_FILENAME = 'custom-elements.md';
 const API_REFERENCE_TITLES = ['## Properties', '## Methods', '## Events', '## Slots'];
 const FOOTER_TITLE = '::footer::';
+const EVENT_WARNING = '::event-warning::';
```
Comment (wattachai-lseg) [inline documents/scripts/element.injector.mjs]: let's rename to 'event-notice'. it's a something you should know but not as serious as a warning.

File: documents/src/templates/event-warning.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<!--
+type: template
+name: event-warning
+-->
+<br>
+Component events will not trigger when properties are set programmatically.
```
Comment (wattachai-lseg) [inline documents/src/templates/event-warning.md]:
```suggestion
Property changed events, such as `open-changed` and `value-changed`, will not be triggered when the properties are updated programmatically.
```

File: packages/elements/src/configuration/elements/configuration.ts [hunk 4]
```diff
@@ -29,6 +29,9 @@ export class Configuration extends BasicElement {
 
   /**
    * Configuration data
+   * @type {Config}
+   * @attr -
```
Comment (wattachai-lseg) [inline packages/elements/src/configuration/elements/configuration.ts:33]: As it's a public API, let's change to `@property({attribute: false})`.

File: packages/elements/src/autosuggest/index.ts [hunk 5]
```diff
@@ -174,7 +174,8 @@ export class Autosuggest extends Overlay {
   public static readonly defaultDebounceRate = 100;
 
   /**
-   * An HTML Element or CSS selector
+   * Set attachment target.
```
Comment (wattachai-lseg) [inline packages/elements/src/autosuggest/index.ts]:
```suggestion
   * Target element to be attached with Autosuggest
```

File: packages/elements/src/button/index.ts [hunk 6]
```diff
@@ -12,9 +12,10 @@ import { VERSION } from '../version.js';
 /**
  * Use button for actions in forms, dialogs,
  * and more with support for different states and styles.
- * @attr {boolean} disabled - Set state to disabled
- * @prop {boolean} [disabled=false] - Set state to disabled
- * @fires active-changed - Fired when `active` property changed by user taps on toggled button. It will not be triggered if `active` state is changed programmatically.
+ * @attr {boolean} disabled - Set disabled state
+ * @prop {boolean} [disabled=false] - Set disabled state
+ *
```
Comment (wattachai-lseg) [inline packages/elements/src/button/index.ts]: remove empty line?
```suggestion
```

File: packages/elements/src/chart/elements/chart.ts [hunk 7]
```diff
@@ -138,9 +138,9 @@ export class Chart extends BasicElement {
   }
 
   /**
-   * List of available chart colors
+   * List of line colors
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/elements/chart.ts]: line colors won't apply to some chart types such as pie & bubble charts.
```suggestion
   * List of available colors for chart styling
```

File: packages/elements/src/clock/index.ts [hunk 8]
```diff
@@ -183,7 +183,8 @@ export class Clock extends ResponsiveElement {
   private _offset = 0;
 
   /**
-   * Get offset value
+   * Shifts displayed time by seconds without changing the original value
```
Comment (wattachai-lseg) [inline packages/elements/src/clock/index.ts]:
```suggestion
   * number of seconds that the display time should be offset from the original value
```

File: packages/elements/src/autosuggest/index.ts [hunk 9]
```diff
@@ -192,7 +193,7 @@ export class Autosuggest extends Overlay {
   public moreResults = false;
 
   /**
-   * Custom text for More Search
+   * Custom text for 'More results'
    * @default More results for {0}
```
Comment (wattachai-lseg) [inline packages/elements/src/autosuggest/index.ts]: What is `{0}`? Is there a way could describe it in a better way?

File: packages/elements/src/combo-box/index.ts [hunk 10]
```diff
@@ -205,7 +202,8 @@ export class ComboBox<T extends DataItem = ItemData> extends FormFieldElement {
   private queryDebouncer = new TimeoutTaskRunner(this._queryDebounceRate);
 
   /**
-   * Control query rate, defaults to 0
+   * Control query rate
```
Comment (wattachai-lseg) [inline packages/elements/src/combo-box/index.ts]:
```suggestion
   * Control query rate in milliseconds
```

File: packages/elements/src/pagination/index.ts [hunk 11]
```diff
@@ -104,7 +105,8 @@ export class Pagination extends BasicElement {
   }
 
   /**
-   * Max page
+   * Maximum page value
```
Comment (wattachai-lseg) [inline packages/elements/src/pagination/index.ts]:
```suggestion
   * Maximum page limit
```

**Review by wattachai-lseg — CHANGES_REQUESTED** [review]

File: documents/src/templates/event-warning.md (hunk 3 above)
Comment (Sarin-Udompanish) [reply documents/src/templates/event-warning.md]: I think the example ( such as `open-changed` and `value-changed** ) is not correct. Events that don't have -changed prefix also not fired when properties are updated programmatically. In another word all events should not be triggered when the properties are updated programmatically.

File: packages/elements/src/configuration/elements/configuration.ts (hunk 4 above)
Comment (Sarin-Udompanish) [reply packages/elements/src/configuration/elements/configuration.ts:33]: Let's do that in another story as It not related to the documentation. The reason that I added @attr - is a workaround to make the api-anlyzer to generate `attribute` column.

File: packages/elements/src/button/index.ts (hunk 6 above)
Comment (Sarin-Udompanish) [reply packages/elements/src/button/index.ts:17]: We tend to leave an empty line to seperate between event and property section.

File: packages/elements/src/autosuggest/index.ts (hunk 9 above)
Comment (Sarin-Udompanish) [reply packages/elements/src/autosuggest/index.ts]: That the actual default value of this property.

Comment (wattachai-lseg) [reply packages/elements/src/autosuggest/index.ts]: it comes from this phrasebook? Maybe we could change it to `More results for {query}` instead? It's still a source code but `query` is definitely better than `0`
```javascript
const translations = {
  LOADING: 'Loading suggestion items',
  MORE_RESULTS: 'More results for <mark>{ query }</mark>'
};
````

File: packages/elements/src/configuration/elements/configuration.ts (hunk 4 above)
Comment (wattachai-lseg) [reply packages/elements/src/configuration/elements/configuration.ts:33]: I understand that this is clearly out of scope. Yet, it's a tiny change. Can't we piggyback on this PR?

File: documents/src/templates/event-warning.md (hunk 3 above)
Comment (wattachai-lseg) [reply documents/src/templates/event-warning.md]: `add-attach-target-events` of Autosuggest component fires when `attach` prop is updated programmatically. We need to find a way to handle this exception and I'm afraid there could be more.

File: documents/scripts/element.injector.mjs (hunk 1 above)
Comment (wattachai-lseg) [reply documents/scripts/element.injector.mjs]: resolving this as it's no longer relevant.

File: documents/scripts/element.injector.mjs (hunk 2 above)
Comment (wattachai-lseg) [reply documents/scripts/element.injector.mjs]: resolving this as it's no longer relevant.

File: documents/src/templates/event-warning.md (hunk 3 above)
Comment (wattachai-lseg) [reply documents/src/templates/event-warning.md]: resolving this as it's no longer relevant.

File: packages/elements/src/configuration/elements/configuration.ts (hunk 4 above)
Comment (wattachai-lseg) [reply packages/elements/src/configuration/elements/configuration.ts:33]: new issue created for this issue: https://jira.refinitiv.com/browse/ELF-2316

File: packages/elements/src/button/index.ts [hunk 12]
```diff
@@ -12,9 +12,9 @@ import { VERSION } from '../version.js';
 /**
  * Use button for actions in forms, dialogs,
  * and more with support for different states and styles.
- * @attr {boolean} disabled - Set state to disabled
- * @prop {boolean} [disabled=false] - Set state to disabled
- * @fires active-changed - Fired when `active` property changed by user taps on toggled button. It will not be triggered if `active` state is changed programmatically.
+ * @attr {boolean} disabled - Set disabled state
+ * @prop {boolean} [disabled=false] - Set disabled state
+ * @fires active-changed - Fired when `active` property changed by user taps on toggled button.
```
Comment (wattachai-lseg) [inline packages/elements/src/button/index.ts]: I could confirm that the event doesn't fire when `active` is updated programmatically.  Why did we change it?

File: packages/elements/src/chart/elements/chart.ts [hunk 13]
```diff
@@ -138,9 +138,9 @@ export class Chart extends BasicElement {
   }
 
   /**
-   * List of available chart colors
+   * List of available colors for chart styling
    * @type {string[]}
-   * @returns {string[]}List of available chart colors
+   * @returns {string[]} List of line colors
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/elements/chart.ts]:
```suggestion
   * @returns {string[]} List of available colors for chart styling
```

File: packages/elements/src/select/index.ts [hunk 14]
```diff
@@ -167,7 +167,7 @@ export class Select extends ControlElement implements MultiValue {
    * @readonly
    */
   @property({ type: String, attribute: false })
-  public get label(): string {
+  public get label(): string | undefined {
```
Comment (wattachai-lseg) [inline packages/elements/src/select/index.ts:170]: I can't find any scenarios when this could be `undefined`.

Comment (wattachai-lseg) [reply packages/elements/src/select/index.ts:170]: my  bad, this would be `undefined` when labels is an empty array.

**Review by wattachai-lseg — DISMISSED** [review]

File: packages/elements/src/appstate-bar/index.ts [hunk 15]
```diff
@@ -46,7 +46,7 @@ export class AppstateBar extends BasicElement {
   public heading = '';
 
   /**
-   * (optional) Type of state bar. Supported value are `info`, `highlight`.
+   * Type of state bar. Supported value are `info`, `highlight`.
```
Comment (wsuwt) [inline packages/elements/src/appstate-bar/index.ts]: Supported values are already mentioned in type column.

```suggestion
   * Type of state bar.
```

**Review by wsuwt — APPROVED** [review]

**Pull request merged**